### ISSUE TITLE
Calibre-web not updating changes made to metadata.db (e.g. by Calibre app)

### ISSUE THREAD

**Issue by Linkandzelda** [janeczku/calibre-web#1557]
**Describe the bug/problem**
Calibre-web is not reading changes to the metadata.db file when it's been changed.

**To Reproduce**
1. Launch the calibre web server
2. Use Calibre on another machine to add a book or edit a book
3. Try to find the book in Calibre-web, it won't be there until restarting the server or clicking "Reconnect Calibre Database"

**Expected behavior**
1. Launch the calibre web server
2. Use Calibre on another machine to add a book or edit a book
3. The new book will appear in the library without a server restart

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Python version: Python 3
 - Calibre-Web version: 1a1d105f
 - Docker container: None
 - Special Hardware: None
 - Browser: Chromium

**Additional context**
I'm using Nextcloud to sync the database to the server, and the client runs on the PC where I access Calibre itself.
Calibre-web is accessed through nginx SSL, which I tested the problem without going through nginx and accessing directly port 8083.

It's worth noting that I tested if this is supposed to work or not by moving the metadata.db file out of the server's copy of the library and tried browsing calibre-web to see if would complain of not finding the db file, and I was still able to browse my library without the presence of metadata.db file which the server started with. This indicates that through usual operation of calibe-web (loading pages, loading a book), it makes no attempts to refresh it's database.

**Comment by Linkandzelda** [comment]
I added code to reload the database file if it's been modified on disk. In `cps/web/py` added the following to import time module:

```
import time
```

Then a way to store the last modified time after the import of `config`:
```
config.calibre_db_last_modified = time.time()
```

Then at the end of the `before_request()` function:
```
    calibre_db_last_modified = float(os.path.getmtime(config.config_calibre_dir))

    if calibre_db_last_modified > config.calibre_db_last_modified:
        config.calibre_db_last_modified = calibre_db_last_modified
        log.debug("Reload database change")
        calibre_db.setup_db(config, ub.app_DB_path)

```

I have not tested every functionality of calibre-web after the change, and this is just a quick patch to get the functionality I wanted.

**Comment by PeterJT** [comment]
I'd like to see this functionality as well. My use case is Calibre on a PC; calibre-web on a docker image on QNAP. Feefilesync replicating the calibre library to QNAP when changes are detected.

**Comment by zahroc** [comment]
Similar setup here...running Calibre on workstation and syncing library to Raspberry Pi. Reconnecting or Restarting always fixes the issue.

**Comment by zisco67** [comment]
Any news on that issue? Why not use @Linkandzelda's approach in production?

**Comment by OzzieIsaacs** [comment]
I can't reproduce the issue. 
I saw a lot of trouble with users trying to copy/replace metadata.db files in the background of calibre-web and wondering why it's not working, so I don't want to encourage them to do so.

Calibre-web is not made for replacing metadata.db while running and it will not in the future

There is a reconnect route (button in admin section) where you can manually reconnct your metadata.db file if you want and also do it automatically, please use this function

**Comment by Linkandzelda** [comment]
I know this issue has been closed, but I just wanted to comment to say I don't agree with @OzzieIsaacs viewpoints. "Just click the button" is not acceptable for 2021 for a functionality which is more than expected behaviour. It reminds of the old "It's now safe to turn off your computer" message. These days everything "just works" the way one would expect it to. I personally expected that it would update itself at least periodically but **it does not**. That was proven by me completely removing the metadata.db file from my server and calibre-web still happily functioned as if it was still there. That shows it makes no attempt to read it after it's been started. Imagine having to restart MySQL every time someone inserted or updated something? It's just not adequate.

I'm happy with the few lines of code that I added to make this functionality work. And it seems trivial to create a better implementation than what I did. If that's not going to happen, then it's just a bit of a shame.
